### ISSUE TITLE
PopSlider tool and IdicatorDisplay component readyEvents fired before data is loaded

### ISSUE THREAD

**Issue by ambodi** [Gapminder/vizabi#64]
```Popslider``` tool has a logical bug that in https://github.com/Gapminder/vizabi/tree/feature/multi-csv-reader/preview_pages/_examples-reader is showing itself by throwing ```ready``` events and calling ```modelReady``` of both ```pop-slider``` tool and ```indicator-display``` component before the data is loaded for each model.

**Comment by ambodi** [comment]
@arthurcamara1 any updates on this? A big blocker for me.

**Comment by arthurcamara1** [comment]
For popslider, the `ready` event is being called correctly. The problem is just `modelReady`, which is not, but using the `ready` event with `model_binds` [like this](https://github.com/Gapminder/vizabi/blob/develop/src/components/_examples/indicator-display/indicator-display.js#L30) would solve it and unblock it.

With the `multi-csv` example, however, the problem is different. I investigated today and saw that `ready` is not being called *at all* when using `multi-csv-reader`, that's why one works and the other doesn't. The problem is that the reader is not returning a promise correctly at all times, therefore it keeps waiting (loading). There's an else condition missing here:

```
javascript
if (query_type === 'geo.name') {
     _this.selectCategories(query, path, query_num, meta_callback);
} else if (_this.isQueryIndicator(indicators, query_type)) {
    _this.selectIndicators(query, path, query_type, query_num, meta_callback);
}
// what happens if it's not geo.name nor a query indicator?
// meta_callback is never called
// "death_rate" falls in that case
// then the reader loads forever
```

Src: https://github.com/Gapminder/vizabi/blob/feature/multi-csv-reader/src/readers/multi-csv.js#L47

In your example, `death_rate` doesn't fall into neither of those if's, and the reader never considers the promise done. 

I simply changed the state in the example page to hook to something else other than death_rate (in this case, I used pop) and it worked with the multicsv reader: [see it in action here](https://dl.dropboxusercontent.com/u/4933279/gapminder/example_multicsv/preview_pages/_examples-reader/multicsv-reader.html)

So I guess it's just a matter of fixing the reader so that it doesn't stay in a loop. Let me know if that was helpful :+1:
